chart/templates/deployment-app.yaml

```diff
@@ -13,8 +13,12 @@ spec:
       {{- include "ots.selectorLabels" . | nindent 6 }}
   template:
     metadata:
-      {{- with .Values.podAnnotations }}
       annotations:
+        # used to detect config changes to trigger update
+        {{- $configEnv := .Values.env }}
```

Comment:
ENV changes automatically trigger a replace, this is not needed.

Reply:
Shouldn't a Env Var trigger a new pod to be created?

Reply:
It does, no annotation needed… An `env: []` entry changes the pod-spec requiring a re-create of the pod (therefore a replacement according to `strategy`).